examples/motoko_examples/whoami/test/test.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    run_tests,
+    Test
+} from 'azle/test';
+import { execSync } from 'child_process';
+import { createActor } from './dfx_generated/whoami';
+import { Ed25519KeyIdentity } from '@dfinity/identity';
+import { SignIdentity } from '@dfinity/agent';
+
+function createIdentity(seed: number): SignIdentity {
+    const seed1 = [seed, ...new Array(31).fill(0)];
+    return Ed25519KeyIdentity.generate(Uint8Array.from(seed1));
+}
+
+const installationPrincipal = execSync(`dfx identity get-principal`).toString().trim();
+
+const identity = createIdentity(1);
+const callingPrincipal = identity.getPrincipal().toString();
+
+const canisterId = 'rrkah-fqaaa-aaaaa-aaaaq-cai'
+
+const whoami_canister = createActor(
+    canisterId, {
+        agentOptions: {
+            host: 'http://127.0.0.1:8000',
+            identity
+        }
+    }
+);
+
+const tests: Test[] = [
+    {
+        name: 'clear canister memory',
+        prep: async () => {
+            execSync(`dfx canister uninstall-code whoami || true`, {
+                stdio: 'inherit'
+            });
+        }
+    },
+    {
+        // TODO hopefully we can get rid of this: https://forum.dfinity.org/t/generated-declarations-in-node-js-environment-break/12686/16?u=lastmjs
+        name: 'waiting for createActor fetchRootKey',
+        wait: 5000
+    },
+    {
+        name: 'deploy',
+        prep: async () => {
+            execSync(`dfx deploy --argument '(principal "${installationPrincipal}")'`, {
```

Comment:
I feel like it would be less confusing if we had an `installationPrincipal` and a `someonePrincipal` to match what the Motoko example is doing. As we've talked about it I felt like it was confusing to follow when we were talking about the installer and when we were just talking about the argument being passed in (someone in the Motoko example).

Reply:
I've created a new `someonePrincipal` so that there isn't confusion between these two.